src/gluonts/dataset/pandas.py

```diff
@@ -238,109 +300,59 @@ def infer_freq(index: pd.Index) -> str:
     return freq
 
 
-def extract_dynamic_array(
-    df: pd.DataFrame, col_names: Union[str, Collection[str]]
-) -> np.ndarray:
-    return df[col_names].values.transpose()
-
-
-def extract_static_array(
-    df: pd.DataFrame, col_names: Union[str, Collection[str]]
-) -> np.ndarray:
-    if isinstance(col_names, str):
-        return df[col_names].values[:1]
-    return df[col_names].values[0, :]
+def remove_last_n(n: int, array: np.ndarray) -> np.ndarray:
+    """
+    Return a new array with last ``n`` elements removed from the
+    trailing axis, if ``n`` is positive, and the array itself otherwise.
+    """
+    if n <= 0:
+        return array
+    return array[..., :-n]
 
 
-def as_dataentry(
-    data: pd.DataFrame,
-    target: Union[str, List[str]],
-    timestamp: Optional[str] = None,
-    feat_dynamic_real: List[str] = [],
-    feat_dynamic_cat: List[str] = [],
-    feat_static_real: List[str] = [],
-    feat_static_cat: List[str] = [],
-    past_feat_dynamic_real: List[str] = [],
-) -> DataEntry:
+def split_numerical_categorical(
```

Comment:
I think I would remove this function together and just do this inline:

```py
self._static_real = self.static_features[
    [col_name for col_name, col in df.items() if col.dtype == "category"]
]
```

And probably even better:

```py
self._static_real = static_features.select_dtypes("number")
self._static_cats = static_features.select_dtypes("category")
```